daemon/SecretsImpl/secrets.cpp

```diff
@@ -2030,6 +2110,29 @@ void Daemon::ApiImpl::SecretsRequestQueue::handleFinishedRequest(
             }
             break;
         }
+        case QueryLockStatusRequest: {
+            Result result = request->outParams.size()
+                    ? request->outParams.takeFirst().value<Result>()
+                    : Result(Result::UnknownError,
+                             QLatin1String("Unable to determine result of QueryLockStatusRequest request"));
+            if (result.code() == Result::Pending) {
+                // shouldn't happen!
+                qCWarning(lcSailfishSecretsDaemon) << "QueryLockStatusRequest:" << request->requestId << "finished as pending!";
+                *completed = true;
+            } else {
+                bool locked = request->outParams.size()
+                        ? request->outParams.takeFirst().value<bool>()
+                        : true;
+                if (request->isSecretsCryptoRequest) {
+                    asynchronousCryptoRequestCompleted(request->cryptoRequestId, result, QVariantList() << locked);
+                } else {
+                    request->connection.send(request->message.createReply() << QVariant::fromValue<Result>(result)
+                                                                            << QVariant::fromValue<bool>(locked));
+                }
+                *completed = true;
```

Comment:
Both branches have *completed = true => could be moved before break

Reply:
will fix